atlp-rwanda/Dominators-Phantom-BE
# Ft sign in to phantom

#### What does this PR do?
* Sign in to phantom
* Administrator to use pre-configured login credentials

#### Description of Task to be completed?
* Sign in to phantom where the Administrator will use pre-configured login credentials.
* Have the following endpoint working
`POST /api/v1/user/login`

#### How should this be manually tested?
* `git pull` this branch 
* run `npm install`
* check env.example file and add JWT_SECRET and JWT_EXPIRES_IN to your .env file
* Do migrations with `npm run migrate`
* run `npx sequelize-cli db:seed:all` to create pre-configured credentials for seeders
* run `npm run dev`
* Test the route `POST /api/v1/user/login` in POSTMAN or any other platform for using API

#### Any background context you want to provide?
1. I used the JWT package for hashing the password stored in the Database
2. Documentation can be accessed at `/api-docs`

#### What are the relevant pivotal tracker stories?
*  **The driver, administrator, and operator** should be able to sign in to the app by providing their email & password.
* The API should return a JW token on a successful login.
* The **administrator** should be able to sign in to the app using pre-configured login credentials.
* Appropriate error messages are displayed on unsuccessful attempts

#### Questions:
No.
